x-pack/plugins/apm/server/routes/service_groups/lookup_services.ts

```diff
@@ -14,20 +16,24 @@ import {
 } from '../../../common/elasticsearch_fieldnames';
 import { ProcessorEvent } from '../../../common/processor_event';
 import { Setup } from '../../lib/helpers/setup_request';
-import { MAX_NUMBER_OF_SERVICE_GROUPS } from '../../../common/service_groups';
 
 export async function lookupServices({
   setup,
+  uiSettingsClient,
   kuery,
   start,
   end,
 }: {
   setup: Setup;
+  uiSettingsClient: IUiSettingsClient;
   kuery: string;
   start: number;
   end: number;
 }) {
   const { apmEventClient } = setup;
+  const maxNumberOfServices = await uiSettingsClient.get<number>(
```

Comment:
@younesmln this is the call you can parallelise with `setupRequest`, awaiting the plugin's start services should be instant, but this one actually goes to Elasticsearch. It means you have to lift this call out of `lookupServices` and into the route handler.